src/executor/query/DataCollectExecutor.cpp

```diff
@@ -249,5 +253,80 @@ Status DataCollectExecutor::collectMultiplePairShortestPath(const std::vector<st
     return Status::OK();
 }
 
+Status DataCollectExecutor::collectPathProps(const std::vector<std::string>& vars) {
+    DataSet ds;
+    ds.colNames = std::move(colNames_);
+    DCHECK(!ds.colNames.empty());
+    // 0: vertices's props, 1: Edges's props, 2: Paths without props
+    DCHECK_EQ(vars.size(), 3);
+
+    std::unordered_map<Value, Vertex> vertexMap;
+    auto vIter = ectx_->getResult(vars[0]).iter();
+    DCHECK(vIter->kind() == Iterator::Kind::kProp);
+    for (; vIter->valid(); vIter->next()) {
+        auto vertex = vIter->getVertex();
+        if (!vertex.isVertex()) {
+            continue;
+        }
+        auto vertexVal = vertex.getVertex();
+        auto vid = vertexVal.vid;
+        vertexMap.insert(std::make_pair(vid, std::move(vertexVal)));
+    }
+
+    std::unordered_map<std::string, Edge> edgeMap;
+    auto eIter = ectx_->getResult(vars[1]).iter();
+    DCHECK(eIter->kind() == Iterator::Kind::kProp);
+    for (; eIter->valid(); eIter->next()) {
+        auto edge = eIter->getEdge();
+        if (!edge.isEdge()) {
+            continue;
+        }
+        auto edgeVal = edge.getEdge();
+        auto srcVid = edgeVal.src;
+        auto dstVid = edgeVal.dst;
+        auto type = edgeVal.type;
+        auto ranking = edgeVal.ranking;
+        auto edgeKey = folly::stringPrintf(
+            "%s%s%d%ld", srcVid.toString().c_str(), dstVid.toString().c_str(), type, ranking);
```

Comment:
If here has edge1: `"123"-[like]->"456"`, and  edge2: `"12"-[like]->"3456"`, the are the same key